lessons/codap-transformers1/langs/en-us/pages/codap-filter-transformer.adoc

```diff
@@ -0,0 +1,89 @@
+= Filter
+
+++++
+<style>
+body.workbookpage td .autonum::after { content: ')' !important; }
+/* Cram everything to the top instead of distributing space equally */
+#content { display: block !important; }
+</style>
+++++
+
+Make sure you’re logged into the @starter-file{animals} in CODAP. Select the Plugins icon, then choose Transformers.
+
+
+@vspace{1ex}
+
+== Create, Apply, and Save a Filter Transformer (Step by Step)
+
+[cols="5, 10", frame="none", grid="none", stripes="none"]
+|===
+
+|
+@image{../images/filter-ui-annotated.png, 350}
+
+|
+@vspace{1ex}
+
+@n Choose Filter from the drop-down menu that appears (Box 1).
+
+@vspace{1ex}
+
+@n Name the Transformer `filter-is-dog`. Type the name into Box 2 (left).
+
+@vspace{1ex}
+
+@n Click on "Dataset to Filter" to confirm that the Animals Dataset is selected.
+
+@vspace{1ex}
+
+@n The contract's input - or domain - is a Row. Why does that makes sense?
+
+@fitb{}{@ifsoln{When we filter, we look at each row, one by one.}}
+
+@vspace{1ex}
+
+@n The output - or range - is a Boolean. Why does that make sense?
+
+@fitb{}{@ifsoln{We are going to determine if the formula is true or false for each row.}}
+
+@vspace{1ex}
+
+@n A purpose statement explains the purpose of the expression. A good rule for purpose statements is to structure them like this: "The expression *_consumes_* one thing and *_produces_* another thing." With your class, write a purpose statement for this expression. Type it into Box 6.
+
+
+|===
+
+
+@n Enter `Species = "dog"` as the expression (Box 7). Select `Apply Transformer`. What happens? @fitb{}{}
+
+@fitb{}{@ifsoln{A copy of the original dataset is created, with only dogs!}}
+
+@n Try typing `species = "dog"` as the expression (instead of `Species = "dog"`). What happens? @fitb{}{}
+
+@fitb{}{@ifsoln{_An error message appears that says, "Evaluating `species = "dog"` failed: 'species' is unrecognized."_}}
+
+@n What are some other possible reasons you might get an error message for the expression? @fitb{}{}
+
+@fitb{}{@ifsoln{_Incorrect spelling, wrong capitalization, forgetting quotation marks._}}
+
+@n Select "Save Transformer." Describe what happens. Why might it be useful to save a Transformer? @fitb{}{}
+
+@fitb{}{}
+
+
+== More Filtering (On Your Own)
+
+@n Create, save, and apply a Transformer called `filter-is-old` that creates a new dataset with animals older than 5 years. How many rows does the resulting table have?
+
+@fitb{}{@ifsoln{_There are 9 rows._}}
+
```

Comment:
I would expect the solutions here to support teachers with how to write the filter.